database/git_index.ksy

```diff
@@ -0,0 +1,150 @@
+meta:
+  id: git_index
+  title: Git index
+  license: CC0-1.0
+  ks-version: 0.9
+  endian: be
+  encoding: UTF-8
+doc-ref: https://github.com/git/git/blob/main/Documentation/technical/index-format.txt
+seq:
+  - id: header
+    type: header
+  - id: entries
+    type: entry
+    repeat: expr
+    repeat-expr: header.num_entries
+  - id: extensions
+    type: extension
+    repeat: until
+    repeat-until: _io.pos >= _io.size - len_hash
```

Comment:
This seems more natural to me and I'm more certain that it is correct:

```suggestion
    repeat-until: _io.pos + len_hash >= _io.size
```

It follows the idea "if I parse the `checksum` from the current position, I'll reach the EOF condition `_io.pos == _io.size`, or `_io.pos > _io.size` (hypothetically) if there are not even enough bytes for `checksum`".